android-animation-arsenal/src/main/java/com/android_animation_arsenal/views/RevealDialog.java

```diff
@@ -0,0 +1,373 @@
+package com.android_animation_arsenal.views;
+
+import android.animation.Animator;
+import android.animation.AnimatorListenerAdapter;
+import android.animation.AnimatorSet;
+import android.animation.ObjectAnimator;
+import android.animation.PropertyValuesHolder;
+import android.app.Dialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.graphics.drawable.ColorDrawable;
+import android.util.DisplayMetrics;
+import android.view.KeyEvent;
+import android.view.MotionEvent;
+import android.view.View;
+import android.view.ViewAnimationUtils;
+import android.view.WindowManager;
+import android.view.animation.AccelerateInterpolator;
+import android.view.animation.DecelerateInterpolator;
+
+/**
+ * Created by angelsolis on 12/9/15.
+ */
+public class RevealDialog extends Dialog
+{
+
+    private int duration = 500;
+    private View contentView;
+    private View targetView;
+    private boolean hasTarget;
+    private boolean isReveal;
+    private boolean isAnimating;
+
+    public RevealDialog(Context context)
+    {
+        super(context);
+        getWindow().clearFlags(WindowManager.LayoutParams.FLAG_DIM_BEHIND);
+        hasTarget = false;
+        isAnimating = false;
+        isReveal = false;
+    }
+
+
+    @Override
+    public boolean onTouchEvent(MotionEvent event)
+    {
+        if(event.getAction() == MotionEvent.ACTION_DOWN && ! isAnimating)
+        {
+            if(hasTarget)
```

Comment:
Target is the view that you want the dialog to be animated with.. In this case the user sets the target he wants to animate with dialog.. if there is no target view then it will just do a simple reveal